lib/proxymode.pm

```diff
@@ -0,0 +1,142 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+package proxymode;
+# Summary: proxymode: The basic lib for using proxy mode to connect or operation with physical machine
+# Maintainer: John <xgwang@suse.com>
+
+use base 'basetest';
+use testapi;
+use strict;
+
+our $SLAVE_SERIALDEV = 'proxyserial';
+
+sub switch_power {
+    my ($ipmi_machine, $ipmi_user, $ipmi_pass, $ipmi_status) = @_;
+    $ipmi_pass   //= 'ADMIN';
+    $ipmi_user   //= 'ADMIN';
+    $ipmi_status //= 'off';
+    die "Variable ipmi_machine is invalid in function restart_host!" unless $ipmi_machine;
+    my $ipmitool = "ipmitool -H " . $ipmi_machine . " -U " . $ipmi_user . " -P " . $ipmi_pass . " -I lanplus ";
+    select_console 'log-console';
```

Comment:
I understand, but you don't need to put `select_console` in `switch_power`. You could put it before your first call to `switch_power` in `restart_host`, then you are decoupling this function from a particular console and saving a remote procedure call to the backend. More importantly, it is then easy for me to see that you are switching to another console to do some work then switching back again.